### PR TITLE
Jofultz kv log

### LINKED ISSUE #231: Update KeyVault to App Log Level Mapping
## Description:
The SDK team is looking into disabling values for KeyVault Secrets.  If that change happens then we can change the mapping to include BODY_AND_HEADERS or BODY as a valid value as secret values would not show up in logs.

This will require a new mapping and that will need to be updated in the code and documentation.

- References #230

### PR DESCRIPTION
**Pull request by jofultz**
## Type of PR

- [ ] Documentation changes
- [X] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

### Purpose of PR
* Added setting to logback.xml and log4j2.properties to reduce overall logging
* Added KeyVaultSecretsLogPolicy.java to redact secret values

### Validation
- [X] Unit tests updated and ran successfully
- [ ] Update documentation or issue referenced above

### Issues Closed or Referenced

- Closes #231

### CODE REVIEW

File: src/main/java/com/cse/helium/app/services/keyvault/KeyVaultSecretsLogPolicy.java [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+  private final Set<String> allowedQueryParameterNames;
+  private final boolean prettyPrintBody;
+  private Pattern secretsPattern;
+
+  /**
+   * Creates an KVSecretsLoggingPolicy with the given log configurations.
+   *
+   * @param httpLogOptions The HTTP logging configuration options.
+   */
+  public KeyVaultSecretsLogPolicy(HttpLogOptions httpLogOptions) {
```
Comment (jyotsnaravikumar) [inline src/main/java/com/cse/helium/app/services/keyvault/KeyVaultSecretsLogPolicy.java:54]: Good Idea to switch to custom policy!

File: src/main/java/com/cse/helium/app/services/keyvault/KeyVaultService.java [hunk 2]
```diff
@@ -94,16 +92,18 @@ public KeyVaultService(IEnvironmentReader environmentReader)
...
         .vaultUrl(getKeyVaultUri())
         .credential(credential)
+        .addPolicy(getKvLogPolicy())
         .buildClient();
 
     secretAsyncClient = new SecretClientBuilder()
         .vaultUrl(getKeyVaultUri())
         .credential(credential)
-        .httpLogOptions(new HttpLogOptions().setLogLevel(logDetailLevel))
+        .addPolicy(getKvLogPolicy())
```
Comment (jyotsnaravikumar) [inline src/main/java/com/cse/helium/app/services/keyvault/KeyVaultService.java:106]: +1

File: src/main/java/com/cse/helium/app/services/keyvault/KeyVaultService.java [hunk 3]
```diff
@@ -119,24 +119,33 @@ public KeyVaultService(IEnvironmentReader environmentReader)
...
+
+      logLevel = HttpLogDetailLevel.HEADERS;
 
-      returnLevel = HttpLogDetailLevel.HEADERS;
     } else if (currentLogLevel == Level.WARN 
           || currentLogLevel == Level.ERROR
           || currentLogLevel == Level.FATAL) {
 
-      returnLevel = HttpLogDetailLevel.BASIC;
+      logLevel = HttpLogDetailLevel.BASIC;
```
Comment (jyotsnaravikumar) [inline src/main/java/com/cse/helium/app/services/keyvault/KeyVaultService.java:141]: BASIC by definitionsLogs everything in BASIC, plus all the request and response headers. Should it apply to WARN as well?

Comment (jofultz) [reply src/main/java/com/cse/helium/app/services/keyvault/KeyVaultService.java:141]: It does apply to WARN based on the code about.  Also, it doesn't log as you note, I placed a matrix of what is logged in #230 .

**Pull request merged**